server/utils.py

```diff
@@ -55,40 +56,54 @@ def ensure_topic(topic_name: str):
     bootstrap_servers = settings.kafka_bootstrap_servers
     retention_ms = settings.kafka_retention_ms
 
-    admin_client = KafkaAdminClient(
-        bootstrap_servers=bootstrap_servers,
-        client_id="topic_creator",
-        api_version=(2, 5, 0),
-    )
+    async def _ensure_topic():
+        admin_client = AIOKafkaAdminClient(
+            bootstrap_servers=bootstrap_servers,
+            client_id="topic_creator",
+        )
 
-    topic = NewTopic(
-        name=topic_name,
-        num_partitions=1,
-        replication_factor=1,
-        topic_configs={"retention.ms": str(retention_ms)},
-    )
+        try:
+            await admin_client.start()
+            topic = NewTopic(
+                name=topic_name,
+                num_partitions=1,
+                replication_factor=1,
+                topic_configs={"retention.ms": str(retention_ms)},
+            )
+
+            try:
+                await admin_client.create_topics([topic])
+            except TopicAlreadyExistsError:
+                # we shouldn't hit this case when KAFKA_CFG_AUTO_CREATE_TOPICS=false unless there is a legitimate name collision, so should raise here after testing
```

Comment:
did you test this and verify whether we can raise here?

Reply:
I think we should be able to, but deployed envs don't even use this setting, so not worth changing now imo